lib/v2/components/scanner/scanner_screen.dart

```diff
@@ -25,6 +25,10 @@ class ScannerScreen extends StatefulWidget {
     _scannerBloc.add(ShowLoading());
   }
 
+  void stop() {
+    _scannerBloc.add(Stop());
+  }
+
   void resultCallBack(String scanResult) {
```

Comment:
Are these methods necessary?

- `void showError(String errorMessage)` I don't see it is being called in SendScannerScreen or elsewhere.
- `void resultCallBack(String scanResult)`  I don't see it is being called in SendScannerScreen or elsewhere. 
Actually this callback `final Function resultCallback;` is called in `widget.resultCallBack(scanResult);` also `Function resultCallback` could be a more specific callback by using one of the typedef of flutter: `ValueSetter<String> onResult` or `AsyncValueSetter<String> onResult`, an asynchronous version.

Reply:
error is not being used. removed.

Reply:
i havent used ValueSetter<String> onResult or AsyncValueSetter<String> onResult but seems like the right approach here